Auto-completeing objectives!

Merlin1230 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Adds objectives that auto-complete. Currently, the objectives are all stored in txt files in strings/objectives/antag.
currently, I need the following objectives:
- [ ] Wizard Objectives 
- [ ] Changeling Objectives
- [ ] Traitor Objectives
- [ ] Blood Brother Objectives

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Having these objectives should encourage people to focus less on getting a greentext, and they also help with getting ideas.
## Changelog
:cl:
add: Antags now get objectives that auto-complete in order to help them with getting ideas!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Merlin1230 committed:
Auto-completeing objectives! almost!

filled the empty lists
wizard got 4
ling got 1
BB got 1
General got 1

allowed admins to add these objectives.
you won't be able to choose which objective,
but you will be able to choose the list it pulls from
and force it to pull from that list
instead of it being a 50/50 to pull from that list or general.

also, we're getting the band back together.

Lets do it like this instead

changed the wording a bit

forgot not all antags do it like BB

MORE

Apply suggestions from code review

Co-authored-by: Kapu1178 <75460809+Kapu1178@users.noreply.github.com>

uses the objective's line number to give out gear

updated comment

Kapu1178 commented on code/game/gamemodes/objective.dm:
```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+/datum/objective/auto_complete/proc/make_random_objective(antag_type = null, force_random_check = null)
+	var/objective
+	if((prob(50) || force_random_check) && antag_type)
+		switch(antag_type)
+			if("traitor") objective = pick(world.file2list("strings/objectives/antag/traitor.txt"))
+			if("wizard") objective = pick(world.file2list("strings/objectives/antag/wizard.txt"))
+			if("changeling") objective = pick(world.file2list("strings/objectives/antag/changeling.txt"))
+			if("BloodBrother") objective = pick(world.file2list("strings/objectives/antag/BloodBrother.txt"))
+	if(!objective) objective = pick(world.file2list("strings/objectives/antag/general.txt"))
```
```suggestion
	if(antag_type)
		objective = pick(world.file2list("strings/objectives/antag/[antag_type].txt"))
	objective ||= pick(world.file2list("strings/objectives/antag/general.txt"))
```
im 99% sure this works

Kapu1178 commented on code/game/gamemodes/objective.dm:
```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+	)
+	var/antag = input(admin, "Which objective list do you want to pull from?", "Antag lists") as anything in antag_list
+	var/random_check
+	if(antag == "general") antag = null
```
```suggestion
	if(antag == "general") 
		antag = null
```

No same lines

Kapu1178 commented on code/game/gamemodes/objective.dm:
```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+	var/antag = input(admin, "Which objective list do you want to pull from?", "Antag lists") as anything in antag_list
+	var/random_check
+	if(antag == "general") antag = null
+	if(antag) random_check = input(admin, "Do you wish to force it take from that specific list?", "Random check") in list("Yes", "No")
```
```suggestion
	if(antag) 
		random_check = input(admin, "Do you wish to force it take from that specific list?", "Random check") in list("Yes", "No")
```

Kapu1178 commented on code/game/gamemodes/objective.dm:
```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+	var/random_check
+	if(antag == "general") antag = null
+	if(antag) random_check = input(admin, "Do you wish to force it take from that specific list?", "Random check") in list("Yes", "No")
+	if(random_check == "No") random_check = null
+	make_random_objective(antag, random_check)
```
```suggestion
	make_random_objective(antag, random_check == "No" ? null : random_check)
```

Kapu1178 commented on code/game/gamemodes/objective.dm:
```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+
+/datum/objective/auto_complete/update_explanation_text()
+	. = ..()
+	switch(objective_text)
```
Objectives should probably be associative lists with a simple key so you dont have entire objective strings as switch statements.

zeskorion committed:
fixes extrapolators again (#6620)

Ivniinvi committed:
Stealth Mode announces your stealth CKEY to other admins (#6621)

ss13-beebot committed:
Automatic changelog generation for PR #6620 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #6621 [ci skip]

Mat05usz committed:
Crewmonitor fixes and refactors (#6549)

Co-authored-by: Mat05usz <asdfasdfasdf@asdfasdfasdfvxcxvx.com>

ss13-beebot committed:
Automatic changelog generation for PR #6549 [ci skip]

Ivniinvi committed:
[PORT] Rust Improvements (#6530)

zeskorion committed:
fixes a few wing bugs (#6534)

ss13-beebot committed:
Automatic changelog generation for PR #6530 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #6534 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Jeff-3 committed:
Small fix (#6516)

ss13-beebot committed:
Automatic changelog generation for PR #6516 [ci skip]

Hardly3D committed:
[PORT] Split personality communication (#6536)

ss13-beebot committed:
Automatic changelog generation for PR #6536 [ci skip]

DingleFlop committed:
PubbyStation Atmosia changes [First PR] (#6400)

MNarath1 committed:
Fixes a lot of create_and_destroy runtimes (#6356)

Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6400 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #6356 [ci skip]

PowerfulBacon committed:
Cult bolas no longer affect holy-proof individuals. (#6121)

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>
Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6121 [ci skip]

Kapu1178 committed:
Random Name refactor (#6335)

ss13-beebot committed:
Automatic changelog generation for PR #6335 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
intelligence potion fixy (#6626)

ss13-beebot committed:
Automatic changelog generation for PR #6626 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
fixes lowmemory mode runtimes (#6624)

ss13-beebot committed:
Automatic changelog generation for PR #6624 [ci skip]

zeskorion committed:
nya (#6630)

ss13-beebot committed:
Automatic changelog generation for PR #6630 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Update dualsaber.dm (#6644)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6644 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Chirynnn committed:
Welded firelocks can no longer be opened by IDs (#6657)

ss13-beebot committed:
Automatic changelog generation for PR #6657 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Mat05usz committed:
Fixes fireman carry bug (#6610)

ss13-beebot committed:
Automatic changelog generation for PR #6610 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Update pyrotechnic_reagents.dm (#6660)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6660 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Update human.dm (#6654)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6654 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
You drop your items on death in battle royale, drop items on death trait (#6659)

* Dropping items on death trait

* Reviews

ss13-beebot committed:
Automatic changelog generation for PR #6659 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Update closets.dm (#6668)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6668 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

williamcll committed:
Updates Kilostation Engineering SMES + airlock + science shuttle (#6672)

Co-authored-by: williamcll <williamcll@users.noreply.github.com>

PestoVerde322 committed:
[Mappers QoL] New icons for zones and items ported from /tg/ (#6656)

* aeiou

* New icons for maps and items

* Accidentaly did a small ammount of tomfoolery

* guess what

ss13-beebot committed:
Automatic changelog generation for PR #6656 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Ivniinvi committed:
Logs (#6677)

ss13-beebot committed:
Automatic changelog generation for PR #6677 [ci skip]

mystery3525 committed:
Makes ectoscopic sniffer more obvious, plus fix (#6598)

* updated ectoscopic sniffer

* Removed notice

* you cannot interact with the living

* Move ectosniffer infront of circuit imprinter

ss13-beebot committed:
Automatic changelog generation for PR #6598 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
Slows nuka cola and meth down (#6475)

* balans

* mm yes

* i forgor

ss13-beebot committed:
Automatic changelog generation for PR #6475 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
Refactor/documentation on various headwear (#6507)

* Part 1 of touching headwear

* collectable

* Should be good enough for helmets?

* riot helmet flashlight attachie

ss13-beebot committed:
Automatic changelog generation for PR #6507 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Fixes a spelling mistake (#6679)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6679 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

chloecat0 committed:
fix wooden walls (#6683)

ss13-beebot committed:
Automatic changelog generation for PR #6683 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
Fixes crew monitors and implements global suit sensors list (#6645)

* e

* nanites too

* delete fixes

* nanites logic fixes

* @bacon

* Update code/modules/mob/living/carbon/human/human.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6645 [ci skip]

PowerfulBacon committed:
Magical / Holy EMP + explosions are protected by anti-magic (#6123)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6123 [ci skip]

PowerfulBacon committed:
Changeling ability rebalancing (#6129)

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>
Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6129 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Fixes de-sword runtime (#6689)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6689 [ci skip]

Archanial committed:
better logic (#6690)

ss13-beebot committed:
Automatic changelog generation for PR #6690 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Commit:
Ports aiming component from NSV13 (#5894)

Co-authored-by: Kapu1178 <75460809+Kapu1178@users.noreply.github.com>
Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>
Co-authored-by: Ben <bbreslicH@gmail.com>

ss13-beebot committed:
Automatic changelog generation for PR #5894 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Ivniinvi committed:
RWalls are no longer rusted by default (#6693)

PowerfulBacon committed:
Observers can no longer use OOC pre-game (#6687)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6687 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
ugh (#6698)

ss13-beebot committed:
Automatic changelog generation for PR #6698 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Gives pre-game OOC (#6700)

* Gives pre-game OOC

* Update code/modules/client/verbs/ooc.dm

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6700 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
antag_pick (#6703)

ss13-beebot committed:
Automatic changelog generation for PR #6703 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Crossedfall committed:
Forces minimist's resolution to 1.2.6 (#6708)

ss13-beebot committed:
Automatic changelog generation for PR #6708 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
Fixes null is jammed runtime (#6709)

* @powerfulbacon

* morbius

* Update code/modules/clothing/under/_under.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

* Update code/modules/clothing/under/_under.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6709 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

red031000 committed:
medbot voice lines (#6544)

* medbot voice lines

* fix tests

ss13-beebot committed:
Automatic changelog generation for PR #6544 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Fixes Radial Hard Dels (#6712)

* Update radial.dm

* Properly fixes the hard deletes

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6712 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

covertcorvid committed:
Smelter improvements (#6509)

* Smelter improvements

Adds link_id for mappers to link a processor and console. Adds mining point redemption.

* Added ID link to stacking machine

* Simplifies LateInitialize and point redemption (thanks Archanial)

* Fix typo

* Made set_on into a toggle

* The version without a stacker console

* Revert "The version without a stacker console"

This reverts commit 7b6ddfa665c0bbe2b8b01189b91d15a287488228.

* Actually, the other way makes them bring the mats onto the shuttle

ss13-beebot committed:
Automatic changelog generation for PR #6509 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

The-Moon-Itself committed:
Fixes misuse of transform in some animate() calls (#6429)

* waddle fix

* dance_machine matrix stuff

* Dance machine fix

* broader typecast

* atom movable typecast

ss13-beebot committed:
Automatic changelog generation for PR #6429 [ci skip]

PowerfulBacon committed:
Update shuttle_console.dm (#6706)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6706 [ci skip]

chloecat0 committed:
Botany fixes (#6705)

* fix plant naming

* fix icons

* fix obj/item/grown not keeping stats

* don't destroy flower bud

ss13-beebot committed:
Automatic changelog generation for PR #6705 [ci skip]

PowerfulBacon committed:
Update pressure_sensor.dm (#6666)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6666 [ci skip]

imsynthex committed:
Disables the radiation storm "the air begins to grow warm" message (#6713)

* Disabled the rad storm telegraph

* Set telegraph_duration on rad storms back to 400

* Uhhh

ss13-beebot committed:
Automatic changelog generation for PR #6713 [ci skip]

PowerfulBacon committed:
Improves the winter boots no slip (#6653)

* Improves the winter boots no slip

* Update code/modules/clothing/shoes/miscellaneous.dm

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6653 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Mat05usz committed:
Yet another Crewmonitor fix (#6717)

* tested tested tested

* nya

ss13-beebot committed:
Automatic changelog generation for PR #6717 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

mystery3525 committed:
[PORT] Makes HE pipe min temp diff not so low (#6707)

* Makes HE pipe min temp diff not so low

* Update code/modules/atmospherics/machinery/pipes/heat_exchange/he_pipes.dm

Co-authored-by: ike709 <ike709@users.noreply.github.com>

Co-authored-by: Putnam3145 <putnam3145@gmail.com>
Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6707 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Ivniinvi committed:
Adds proximity check to Acid Ooze (#6722)

* adds prox check to acid ooze

* Update code/datums/mutations/actions.dm

Co-authored-by: ike709 <ike709@users.noreply.github.com>

Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6722 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Ivniinvi committed:
power monitors correctly hide cogged apcs (#6723)

ss13-beebot committed:
Automatic changelog generation for PR #6723 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

AgentCitrus committed:
Adds gimmick objectives for traitors (#6506)

* adds some tator gimmick objectives

* changes some objectives, adds some departments

* adds a bunch more objectives, moves generic objectives into a txt file

* gives admins the ability to hand out randomly generated gimmick objectives

* martyr compatible true

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

* cleans up a for loop

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

* uses a define for objective text file

* moves the file define to configuration.dm

* replaced a few objectives, added one new objective

* reverts the for loop commit, adds a new define file for departments

also adds the path to the txt file for the objectives

* changes some objectives

* gimmick objectives now show up as optional instead of successful at roundend

* cleaner code (tm)

* makes it more clear objectives are optional, more optimized code

* optional objective variable, cleaned a condition

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6506 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Fixes chicken nugget box alignment (#6732)

ss13-beebot committed:
Automatic changelog generation for PR #6732 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Mat05usz committed:
e (#6727)

ss13-beebot committed:
Automatic changelog generation for PR #6727 [ci skip]

Rukofamicom committed:
adds missing short_desc (#6730)

Rukofamicom committed:
Tiny Hand Labeler (#6731)

ss13-beebot committed:
Automatic changelog generation for PR #6730 [ci skip]

Archanial committed:
fix (#6735)

ss13-beebot committed:
Automatic changelog generation for PR #6731 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #6735 [ci skip]

Rukofamicom committed:
Hypospray refills work again (#6733)

* fixes hyposprays

Hyposprays are no longer set to a maximum volume of 0 while simultaneously losing all of their reagent container properties.

* Combat hyposprays are not for combat.

Prevents combat hyposprays from being weaponized, as was discussed and approved in a different PR

* Revert "Combat hyposprays are not for combat."

This reverts commit c23d435c114a45ce4be251048c3d684c10a7447d.

ss13-beebot committed:
Automatic changelog generation for PR #6733 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

BrumalLake committed:
ports 5 piping layers from tg (#6039)

Co-authored-by: Ghilker <42839747+Ghilker@users.noreply.github.com>
Co-authored-by: nicbn <contact@nicbn.com>

ss13-beebot committed:
Automatic changelog generation for PR #6039 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ike709 committed:
Ports the ability to disable harddels (#6317)

* Ports the ability to disable harddels

* Update code/__DEFINES/time.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Co-authored-by: ike709 <ike709@github.com>
Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6317 [ci skip]

Hardly3D committed:
New sounds (#6576)

Sounds from https://github.com/tgstation/tgstation/pull/46375

ss13-beebot committed:
Automatic changelog generation for PR #6576 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

MNarath1 committed:
Converts ai tracking towards using signals (#6491)

* converts ai tracking towards using signals

this rewrites ai tracking code from its former spawn(0) while loop to using signals instead behaviors stays mostly consistant except that its a bit faster because there is no sleep proc in there anymore

* small change

* whops almost missed that one

* moves move_camera_by_click to eye object

* small change

* fix and implemention for resolving an edge case

* comment change

* changing proc name and small fix

* removes eyeobj check

i tried the cases where this is supposed to happen but it can't litterally unless some adminbus takes place i don't see any way this happens

* oops

* forgot a space

* puts the return one step higher and inverts the if check

this way i only need a single return instead of two so i guess its better

* fixes argument offset caused by signals

because signals will forward the reference of the signal source and the qdelete signal can also forward the force argument this needs to be offset by the correct ammount

* changed my mind lets actually wrap this in another proc

ss13-beebot committed:
Automatic changelog generation for PR #6491 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
fix (#6741)

ss13-beebot committed:
Automatic changelog generation for PR #6741 [ci skip]

Ivniinvi committed:
bumps min client version (#6726)

ss13-beebot committed:
Automatic changelog generation for PR #6726 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Fixes attachment overlays on guns (#6755)

ss13-beebot committed:
Automatic changelog generation for PR #6755 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

mystery3525 committed:
what (#6760)

ss13-beebot committed:
Automatic changelog generation for PR #6760 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
CTF species whitelist (#6454)

* hair color and stuff + ctf

* whitelist

* undoes the hair part

ss13-beebot committed:
Automatic changelog generation for PR #6454 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Vend-a-tray shards fix + new breaking sound.  (#6729)

* No Shards, New break sound

* Update displaycase.dm

* reduce volume a bit

ss13-beebot committed:
Automatic changelog generation for PR #6729 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Dejaku51 committed:
fix flamethrower runtime (#6774)

ss13-beebot committed:
Automatic changelog generation for PR #6774 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
fix (#6773)

ss13-beebot committed:
Automatic changelog generation for PR #6773 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Removes try catch (#6763)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6763 [ci skip]

PowerfulBacon committed:
acid fix (#6649)

Co-authored-by: PowerfulBacon <>

ss13-beebot committed:
Automatic changelog generation for PR #6649 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ivanmixo committed:
Fixes being able to eat thru gas masks if you're on harm intent (#6781)

ss13-beebot committed:
Automatic changelog generation for PR #6781 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
fix (#6736)

ss13-beebot committed:
Automatic changelog generation for PR #6736 [ci skip]

Penwin0 committed:
Sentinel's Compromise tweaks (#6782)

* compromise

* compromise

* message and oozelings

* true

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6782 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
fixes null runtime (#6771)

Archanial committed:
Removes support for invalid timers (#6734)

* removes support for invalid timers

* better logging?

* ok

* even better logging

ss13-beebot committed:
Automatic changelog generation for PR #6771 [ci skip]

ss13-beebot committed:
Automatic changelog generation for PR #6734 [ci skip]

Archanial committed:
Fixes adding component to qdeleting object runtime (#6747)

* fix

* Update code/game/objects/items/devices/lightreplacer.dm

Co-authored-by: ike709 <ike709@users.noreply.github.com>

* Update code/game/objects/items/stacks/stack.dm

Co-authored-by: ike709 <ike709@users.noreply.github.com>

Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6747 [ci skip]

Rukofamicom committed:
Corrects nitryl to the same calculation stimulum uses. (#6789)

ss13-beebot committed:
Automatic changelog generation for PR #6789 [ci skip]

Hardly3D committed:
[PORT] New sounds when interacting with stuff (#6553)

* Adds clicking sounds, new hypospray sound, makes a bunch of stuff click.

* Adds a speaker option for mining scanner

* Update code/game/machinery/_machinery.dm

Co-authored-by: ike709 <ike709@users.noreply.github.com>

Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6553 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Mat05usz committed:
nya (#6790)

ss13-beebot committed:
Automatic changelog generation for PR #6790 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Archanial committed:
Possibly fixes null client on tgui intialize runtime (#6748)

ss13-beebot committed:
Automatic changelog generation for PR #6748 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

Rukofamicom committed:
Various things involving the Captain's Laser and display cases.  (#6753)

* Cap gun bugfix + PR 1213 reverted

* grumble syntax grumble comma

* update references to the antique gun

* refactors defining unique cell charge

* Fixes attachment overlays on guns

* Implements cases changing access levels at blue alert

* Fixes the display case in the lavaland snow biome

Captain being able to arbitrarily access this case would be a very strange occurrence.

* You can now close displays without access

* Greatly cleans up the code

* One more comment updated

* Cap's gun is an antique again

* variables explained

* vend-a-tray stuff

* Update displaycase.dm

ss13-beebot committed:
Automatic changelog generation for PR #6753 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]

PowerfulBacon committed:
Fixes whiteships and other shuttle ruins not loading (#6667)

* Fixes whiteships and other shuttle ruins not loading

* Updates the mapped shuttles

* Update code/_compile_options.dm

Co-authored-by: Crossedfall <ping@crossedfall.io>

* Update code/modules/shuttle/shuttle.dm

* Update code/modules/shuttle/shuttle.dm

Co-authored-by: PowerfulBacon <>
Co-authored-by: Crossedfall <ping@crossedfall.io>

ss13-beebot committed:
Automatic changelog generation for PR #6667 [ci skip]

PowerfulBacon committed:
Some orbital map improvements + supercruise code changes (#6113)

Co-authored-by: PowerfulBacon <>
Co-authored-by: KubeRoot <kubinator4321@gmail.com>

ss13-beebot committed:
Automatic changelog generation for PR #6113 [ci skip]

ivanmixo committed:
Reworks gentle bluespace/bluespace extract luminescent interaction (#6207)

Co-authored-by: ike709 <ike709@users.noreply.github.com>

ss13-beebot committed:
Automatic changelog generation for PR #6207 [ci skip]

ss13-beebot committed:
Automatic changelog compile [ci skip]
